### DISCUSSION TITLE
Lightweight builds for simple sub-assemblies

### DISCUSSION THREAD

**Discussion by wolffam** [inventree/InvenTree#2830] [General]
My use case is a bit different than most Inventree users.  I use Inventree as an inventory platform for my company to manage the build of an electro-mechanical assembly (50 or so unique parts) where some of those 50 parts are sub assemblies that we assemble ourselves.  Inventree provides part traceability and a hierarchical structure that other platforms were either missing or were implementing with too much friction for our use case.

To that end, some of the sub assemblies that we put together are pretty trivial and requiring a Build Order for those is a bit heavy weight.  We previously used a workflow for those assemblies where we created an empty stock item from the part and then installed stock items into that part.  That functionality was removed when Inventree required "installed stock items" to be serialized.  I understand the motivation there but am curious if there is a workflow that is suggested for those lightweight subassembly builds.

One suggestion I have would be to allow stock items with batch numbers to be installed/uninstalled.  Along similar lines, when installing untracked stock items, it would be beneficial to have the batch number (if available) show in the dialog (see image).  Currently, our technicians have to determine the batch number based on the quantity remaining which is fragile.

![Screen Shot 2022-04-14 at 10 35 36 AM](https://user-images.githubusercontent.com/11801197/163443472-5f6c1fcd-11ca-492c-8337-72126787920f.png)

**Comment by SchrodingersGat** [comment]
I can appreciate that build order process can be "cumbersome". I have already put in a lot of work to streamline it, after feedback from the technical team at my own workplace. 

I do feel that the build order process is the "right" way to handle this, as it enforces a good set of practices, provides improved traceability, etc. But I also agree that it could be simplified for use cases like yours. 

I'd be happy to entertain any suggestions as to how the build order process could be "improved" for your case.

Allowing batch-tracked stock items to be installed might be a good compromise too.

**Reply by wolffam** [reply]
I see what you're saying.  I would agree that the build order route is the way to go and it's easier to just standardize on that from a platform development point of view.

The feature missing for us is that ability to install batched stock items.  For instance, if we have an assembly consistently of four screws and a computer, the computer will be a trackable/serialized component and the screws batched but untracked. 
If a screw breaks or gets lost, we need functionality to install a new screw into that assembly.

I am using a really simple example but our use case requires traceability on all of our parts in an assembly (batch level is acceptable).  

I think the feature we would need is to allow the install of batched stock items at this view:
![Screen Shot 2022-04-15 at 4 17 12 PM](https://user-images.githubusercontent.com/11801197/163652391-19069d2b-1172-4979-affa-1b8725dffb7c.png)

**Reply by SchrodingersGat** [reply]
@wolffam would it be acceptable to allow any part designed as `trackable` to be installed, whether or not it has a serial number, or batch code, or neither?

**Reply by wolffam** [reply]
@SchrodingersGat , that should work for us.  Currently, if a part is trackable, I think a serial number is required so I was working under that assumption.  Removing that constraint and allowing "trackable" parts to be installed would work!

**Reply by sintech** [reply]
Sorry for butting in, but isn't removing trackable constraint solve this issue? Especially for part like screws.

**Reply by wolffam** [reply]
To clarify, I think @SchrodingersGat is suggesting changing the requirements for a part to be trackable.  Currently, a trackable part requires a serial number.  I believe the suggestion is to remove that requirement and then there shouldn't be any needed changes to the install process since that logic could likely stay the same

**Reply by matmair** [reply]
@SchrodingersGat I would at least require a batch code or a serial number so we have some way of cross-checking in the real world.

**Reply by SchrodingersGat** [reply]
My proposal here is that to manually install stock items (child) into other items (parent), the *child* item must be for a *trackable* part, and also have either a *serial number* or *batch code*.

@wolffam this would meet your requirement?

**Reply by wolffam** [reply]
That would indeed meet my requirement.
My only other suggestion that inventree would need is to show the batch number when a user goes to manually install a stock item (the same dialog in the image I posted above).  It would beneficial to add that extra info to the "Allocate Stock" view during the canonical build process too.  Let me know if that needs some further clarification

**Reply by SchrodingersGat** [reply]
Understood. The addition of "viewing batch info" in the modal forms has already been implemented :)
